### PR TITLE
Assignment7

### PR DESCRIPTION
**Pull request by jmplett**

### CODE REVIEW

**Comment by worseTyler** [comment]
I will review this

File: SecretSanta/test/SecretSanta.EndToEnd.Tests/EndToEndTests.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            await Task.WhenAll(
+                Task.Run(
+                    async () => { response = await page.WaitForNavigationAsync(); }), 
+                page.ClickAsync("text=Groups"));
+
+            Assert.IsTrue(response.Ok);
+        }
+
+        [TestMethod]
+        public async Task NavigateToGifts()
```
Comment (worseTyler) [inline SecretSanta/test/SecretSanta.EndToEnd.Tests/EndToEndTests.cs:89]: I feel this can be grouped into one test instead of three but that's my opinion

**Comment by worseTyler** [comment]
Verify that when landing on the homepage, Secret Santa is displayed in the navigation banner ✔
Verify that all 3 pages (Users, Groups, Gifts) can be reached via clicking on the navigation elements ✔
Create a Gift (should assert that the number of gifts after the creation is one greater than the number of gifts before the user was created) ❌
Modify the last gift in the list (should assert that the updated data was successfully saved/displayed to the screen) ❌
Delete the last gift in the list (should assert that the number of gifts after the delete is one less than the number of gifts before the deletion) ❌
All of these tests should be able to be executed without having to manually start the Web and Api projects ✔
Using AlpineJS, convert the CRUD pages for Users to be javascript based instead of MVC based
Index page should fetch the Users via javascript directly from the API ❌
Create page should submit the Users information via javascript directly to the API. Upon successful save, the user should be redirected back to the Users index page ❌
Edit page should submit the Users information via javascript directly to the API. Upon successful update, the user should be redirected back to the Users index page ❌
Delete opration should submit the request directly to the API via javascript and then refresh the Index page via a javascript call directly to the API ❌

While you have things in your typescript, you didn't actually change any of the Gifts view pages to actually use the typscript. Instead they still use all of the asp.net.
You also only test like half of everything that you are supposed to test

**Comment by worseTyler** [comment]
image
I also ran your E2E tests and they all gave me this error

**Review by breaman — COMMENTED** [review]
Nice job on the assignment. I see that you ended up generating the typescript client but then never used it. Since I never covered that in class, not all that concerned, just saw you made the effort and then didn't use it.

**Pull request closed without merging**